File: front/src/app/models/business/mealComponent.ts
import { Aliment } from './aliment';
import { Recipe } from './recipe';
import { DietComponent } from './dietComponent';

// import { Aliment } from './aliment';

export class MealComponent {

    id: number;
    quantity: number;
    dietComponent: DietComponent = null;
    aliment: Aliment = null;
    recipe: Recipe = null;

    constructor(id: number, quantity: number, aliment: Aliment, recipe: Recipe) {
        this.id = id;
        this.quantity = quantity;
        if(aliment !== null){
            this.dietComponent = aliment;
        } else if ( recipe !== null) {
            this.dietComponent = recipe;
        }
        this.aliment = aliment;
        this.recipe = recipe;
    }

    public getId() {
        return this.id;
    }

    /*
    constructor(id: number, quantity: number , aliment: Aliment) {
        this.aliment = aliment;
        this.quantity = quantity;
    }

    public setAliment(aliment: Aliment) {
        this.aliment = aliment;
    }

    public getAliment() {
        return this.aliment;
    }
    */
}

File: dietcare/src/main/java/co/simplon/dietcare/model/MealComponent.java
package co.simplon.dietcare.model;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

@Entity
@Table(name = "mealcomponent")
/*@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@JsonIdentityInfo(
	  	scope=MealComponent.class,
		generator = ObjectIdGenerators.PropertyGenerator.class, 
	  property = "id")*/
public class MealComponent {
	
	@Id
	@GeneratedValue
	@Column(name = "id")
	private Long id;
	
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}
	
	@JsonIgnore
	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "fk_dietcomponent", nullable = false)
	private DietComponent dietComponent;
	
	@JsonIgnore
	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "fk_meal", nullable = false)
	private Meal meal;
	
	@Column(name = "quantity")
	private int quantity;

	public DietComponent getDietComponent() {
		return dietComponent;
	}

	public void setDietComponent(DietComponent dietComponent) {
		this.dietComponent = dietComponent;
	}

	public Meal getMeal() {
		return meal;
	}

	public void setMeal(Meal meal) {
		this.meal = meal;
	}

	public int getQuantity() {
		return quantity;
	}

	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}

}
